lib/assessment/label.py

```diff
@@ -43,6 +44,22 @@ def statically_label_student_work(self, rubric, student_code, student_id, exampl
                     )
                 )
             }
+        
+        results = {"metadata": {"agent": []},"data": []}
+    
+        for row in csv.DictReader(rubric.splitlines()):
+            if 'CFE' in row["Instructions"]:
+                cfe = CodeFeatures()
+                cfe.extract_features(student_code, row)
+                results["metadata"]["agent"] = ["code feature extractor", "static analysis"]
+                results["data"].append({"Label": cfe.assessment,
+                                        "Key Concept": row["Key Concept"],
+                                        "Observations": cfe.features,
+                                        "Reason": row[cfe.assessment] if cfe.assessment else ''
+                                        })
```

Comment:
I like the term "static analysis" for anything that is not using AI (including detecting empty programs/ programs unchanged from starting code), and renaming this "cfe_label_student_work", or something similar

Reply:
Also agree with Dave about not having the CFE enabled in the rubric itself via the "Instructions" column, since that could change the results from AI, unless there is code to remove the Instructions column prior to concatenating w/ the prompt and sending it to OpenAI

Reply:
> Also agree with Dave about not having the CFE enabled in the rubric itself via the "Instructions" column, since that could change the results from AI, unless there is code to remove the Instructions column prior to concatenating w/ the prompt and sending it to OpenAI

To close the loop on one thing, the Instructions column cannot be removed for the purpose of CFE because it is part of our implementation for chain-of-thought (before and after swapping the Reason and Grade column). You'll find an Instructions column in every `standard_rubric.csv` file in S3.